rero_ils/modules/documents/dojson/contrib/marc21tojson/model.py

```diff
@@ -82,46 +82,24 @@ def get_contribution_link(bibid, reroid, id, key, value):
     prod_host = 'mef.rero.ch'
     test_host = os.environ.get('RERO_ILS_MEF_HOST', 'mef.rero.ch')
     mef_url = 'https://{host}/api/'.format(host=test_host)
-    mef_link = None
     try:
         identifier = id[1:].split(')')
-        url = "{mef}mef/?q={org}.pid:{pid}".format(
-            mef=mef_url,
-            org=identifier[0].lower(),
-            pid=identifier[1]
-        )
-        request = requests.get(url=url)
-        if request.status_code == requests.codes.ok:
-            pid = None
-            data = request.json()
-            hits = data.get('hits', {}).get('hits')
-            if hits:
-                idref = hits[0].get('metadata', {}).get('idref')
-                gnd = hits[0].get('metadata', {}).get('gnd')
-                rero = hits[0].get('metadata', {}).get('rero')
-                if idref:
-                    pid_type = 'idref'
-                    pid = idref['pid']
-                elif gnd:
-                    pid_type = 'gnd'
-                    pid = gnd['pid']
-                elif rero:
-                    pid_type = 'rero'
-                    pid = rero['pid']
-            if pid:
-                mef_link = "{url}{pid_type}/{pid}".format(
-                    url=mef_url,
-                    pid_type=pid_type,
-                    pid=pid
-                )
-                mef_link = mef_link.replace(test_host, prod_host)
-        else:
-            error_print('ERROR MEF REQUEST:', bibid, reroid, url,
-                        request.status_code)
-
+        pid_type = identifier[0].lower()
+        pid = identifier[1]
+        if pid_type == 'idref':
+            url = "{mef}idref/{pid}".format(
+                mef=mef_url,
+                pid=pid
+            )
+            request = requests.get(url=url)
+            if request.status_code == requests.codes.ok:
+                return url.replace(test_host, prod_host)
+            else:
+                error_print('WARNING MEF IDREF NOT FOUND:', bibid, reroid, url,
+                            request.status_code)
+                return None
     except Exception as err:
-        error_print('WARNING NOT MEF REF:', bibid, id, key, value, err)
-    return mef_link
+        pass
```

Comment:
Do you need some logging here?

Reply:
No, I don't think so. We only like to get id's prefixes with `(??????)`, in the moment we have only `(IDREF)` (see line 89).